### PR TITLE
feat(route): 为oreno3d路由添加最大抓取页面功能

### PR DESCRIPTION
**Pull request by xueli-sherryli**
## 该 PR 相关 Issue / Involved issue

Close #10481

## 完整路由地址 / Example for the proposed route(s)

```routes
/oreno3d/search/bronya/latest/1
/oreno3d/characters/283/latest/1
/oreno3d/authors/3189/latest/1
/oreno3d/tags/177/latest/1
/oreno3d/origins/3/latest/1
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
为oreno3d路由添加最大抓取页面功能,使得用户可通过参数控制最大爬取网页数量，可与`?limit=`配合使用，并为页面增加了缓存

### CODE REVIEW

File: lib/v2/oreno3d/main.js [hunk 1]
```diff
@@ -14,81 +14,124 @@ const sortRename = {
...
+                        video_name,
+                        authors,
+                        origins,
+                        characters,
+                        tags,
+                        desc,
+                        iwara_link,
+                        oreno3d_link
+                    });
+                    return result;});
```
Comment (TonyRL) [inline lib/v2/oreno3d/main.js:79]: No need to cache this part as it will be included in L81

File: lib/v2/oreno3d/main.js [hunk 2]
```diff
@@ -14,81 +14,124 @@ const sortRename = {
...
+                const title = `${video_name} - ${authors}`;
+                const realData = await ctx.cache.tryGet(title, () => {
```
Comment (TonyRL) [inline lib/v2/oreno3d/main.js:81]:
```suggestion
                const realData = await ctx.cache.tryGet(oreno3d_link, () => {
```
Please use the site link as the cache key

File: lib/v2/oreno3d/main.js [hunk 3]
```diff
@@ -14,81 +14,124 @@ const sortRename = {
...
+    const userUrl = get_user_url(rootUrl, ctx, sort);
+    // 确认最大页数并比较修改
+    const response = await got(userUrl);
+    const $ = cheerio.load(response.data);
+    const maxPageSelector = "div.container > main > div.g-main-list > ul.pagination > li:last-child > a";
+    // 页码判断
+    if ($(maxPageSelector)) {
+        const pageLink = new URLSearchParams($(maxPageSelector).attr('href'));
+        const actualNum = pageLink.get("page");// 获取最大页数
+        if (Number.parseInt(pagelimit) >= Number.parseInt(actualNum)) {pagelimit = actualNum;}
```
Comment (TonyRL) [inline lib/v2/oreno3d/main.js:109]:
```suggestion
        if (parseInt(pagelimit) >= parseInt(actualNum)) {pagelimit = actualNum;}
```

File: lib/v2/oreno3d/main.js [hunk 4]
```diff
@@ -14,81 +14,122 @@ const sortRename = {
...
+                const video_name = sec_data.video_name;
+                const authors = sec_data.authors;
+                const origins = sec_data.origins;
+                const characters = sec_data.characters;
+                const tags = sec_data.tags;
+                const desc = sec_data.desc;
+                const iwara_link = sec_data.iwara_link;
+                const oreno3d_link = sec_data.oreno3d_link;
+                // 打包,缓存HTML
+                const description = await art(path.join(__dirname, 'templates/description.art'), {
```
Comment (TonyRL) [inline lib/v2/oreno3d/main.js:67]: this is synchronous
```suggestion
                const description = art(path.join(__dirname, 'templates/description.art'), {
```

File: lib/v2/oreno3d/main.js [hunk 5]
```diff
@@ -14,81 +14,122 @@ const sortRename = {
...
+    // 页码判断
+    if ($(maxPageSelector)) {
+        const pageLink = new URLSearchParams($(maxPageSelector).attr('href'));
+        const actualNum = pageLink.get("page");// 获取最大页数
+        if (parseInt(pagelimit) >= parseInt(actualNum)) {pagelimit = actualNum;}
+        }
+    else {pagelimit = 1;}
+    // 构造页码数组
+    const linkList = [];
+    for (let i = 0; i < pagelimit; i++) {
```
Comment (TonyRL) [inline lib/v2/oreno3d/main.js:112]:
```suggestion
    for (let i = 1; i < pagelimit; i++) {
```
`${userUrl}` and `${userUrl}&page=1` are the same

Comment (xueli-sherryli) [reply lib/v2/oreno3d/main.js]: but it begins at ${userUrl}&page=1,and the default is 1

Comment (xueli-sherryli) [reply lib/v2/oreno3d/main.js]: ${userUrl}&page=0 is invaild

Comment (TonyRL) [reply lib/v2/oreno3d/main.js]: You are visiting `${userUrl}` and then `${userUrl}&page=1` which has the same content.
image

Comment (xueli-sherryli) [reply lib/v2/oreno3d/main.js]: the first page which not has `&page` is use for check the real num of pages,so I do not know how to solve this promblem

**Pull request closed without merging**